internal/pkg/cli/recorder/recorder.go

```diff
@@ -103,40 +126,51 @@ func (r *Recorder) Run() error {
 func (r *Recorder) processData(mntns uint32) error {
 	log.Printf("Processing recorded data")
 
+	syscallsMap := map[string]bool{}
+	foundMntns := false
 	it := r.SyscallsIterator(r.bpfRecorder)
 	for r.IteratorNext(it) {
 		currentMntns := binary.LittleEndian.Uint32(r.IteratorKey(it))
-		if currentMntns != mntns {
+		if mntns != 0 && currentMntns != mntns {
 			continue
 		}
-		log.Printf("Found process mntns %d in bpf map", mntns)
+		foundMntns = true
+		log.Printf("Found process mntns %d in bpf map", currentMntns)
 
 		syscallsValue, err := r.SyscallsGetValue(r.bpfRecorder, currentMntns)
 		if err != nil {
 			return fmt.Errorf("get syscalls from bpf map: %w", err)
 		}
 
-		syscalls := []string{}
 		for id, found := range syscallsValue {
 			if found != 0 {
 				name, err := r.GetName(libseccomp.ScmpSyscall(id))
 				if err != nil {
 					return fmt.Errorf("get syscall name for id %d: %w", id, err)
 				}
 
-				syscalls = append(syscalls, name)
+				syscallsMap[name] = true
```

Comment:
With `--no-start`, we are taking all mount namespaces into account (we match on the process name only). So this is using a map as a set to avoid duplicates.

Reply:
Thanks for clarifications, it makes sense then.